### PR TITLE
ChickenKyiv/loopback-tutor-intern-10#14 - Fixed bugs on recipe import. changed file names

### PR DESCRIPTION
**Pull request by louisevdb84**
The current import script did not work on recipes.  I fixed a couple of bugs and changed file names to conform with https://github.com/ChickenKyiv/recipe-search-api naming. 

Please confirm whether the import scripts and the helper script  in these files needs to look exactly like https://github.com/ChickenKyiv/recipe-search-api?

### CODE REVIEW

File: server/datasources.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "recipeDS": {
-    "url": "mongodb://heroku_p0dxgncb:gl2rr2bi235aoqfdojelqspjn1@ds147052.mlab.com:47052/heroku_p0dxgncb",
```
Comment (louisevdb84) [inline server/datasources.json]: I added this file by mistake.

File: bin/recipes/001-import.js [hunk 2]
```diff
@@ -67,14 +68,21 @@ async.parallel({
...
-		helper.create(options, Recipe2, (err, data) => {
+		helper.create(options, Recipes2, (err, data) => {
 			console.log(data);
 		});
 
-		options.predata = await Department.find({ where: {
-			name: 'Meat'
-		}});
-		console.log(options.predata);
+		const meatDepartments = Department.get().filter(department => {
```
Comment (atherdon) [inline bin/recipes/001-import.js]: Department.get() will query all departments from database? i mean there can be a lot of information. I think my way was more easy for db system.

File: server/datasources.json (hunk 1 above)
Comment (atherdon) [reply server/datasources.json]: no prob. later i'll explain how we trying to update logic for basic static things like db link, api keys, etc.
it's quite hard sometimes to figure out what database belong to.

File: bin/recipes/001-import.js [hunk 3]
```diff
@@ -30,14 +32,13 @@ let options = {
...
 }
 
-
 async.parallel({
 		users       : async.apply(helper.create, options, Users),
 		recipes     : async.apply(helper.create, options, Recipes),
 		menus       : async.apply(helper.create, options, Menus),
 
 		// for recipes 2
-		ingr-ex     : async.apply(helper.create, options, IngrEx),
```
Comment (louisevdb84) [inline bin/recipes/001-import.js]: It didn't want to accept dashes when I tried to run the import,, so had to remove the dash

File: bin/recipes/001-import.js [hunk 4]
```diff
@@ -67,14 +68,21 @@ async.parallel({
 
 
 		// imported recipes for search data
-		helper.create(options, Recipe2, (err, data) => {
+		helper.create(options, Recipes2, (err, data) => {
```
Comment (louisevdb84) [inline bin/recipes/001-import.js]: Recipe was declared as Recipes2, so I just added the "s"

File: bin/recipes/001-import.js (hunk 3 above)
Comment (atherdon) [reply bin/recipes/001-import.js]: not sure what do you mean by dashes?

File: bin/recipes/001-import.js (hunk 4 above)
Comment (atherdon) [reply bin/recipes/001-import.js]: yep, i think i tried here to have an old version to work (aka Recipes Model) and tried to make works an updated version as Recipe2. Good approach! Maybe your current changes will force me to update the actual db architecture code. it was done 1y ago, so i think i start to know more about what should be improved here

File: server/datasources.json (hunk 1 above)
Comment (atherdon) [reply server/datasources.json]: here we'll address to this improvement: https://github.com/GroceriStar/groceristar/issues/518

File: bin/recipes/001-import.js (hunk 3 above)
Comment (louisevdb84) [reply bin/recipes/001-import.js]: This is the error I get:

image

Comment (atherdon) [reply bin/recipes/001-import.js]: rename it to `ingrEx` or `ingr_ex`. maybe node have some restrictions in variable naming

**Comment by atherdon** [comment]
Reviewing your changes right now. we updated the import, right?
just now sure how i can test it. i make a bin folder very messy. and forgot what i need to call in order to review your changes...

**Comment by atherdon** [comment]
i run this scripts and all is work now - is this what i should do, right?
```
"migrate-recipes": "node bin/recipes/002-automigrate.js",
"import": "node bin/recipes/001-import.js",
```

File: bin/recipes/001-import.js (hunk 3 above)
Comment (louisevdb84) [reply bin/recipes/001-import.js]: cool, it was already renamed to ingrEx in this pull request. I just changed it back in my copy for demonstration purposes.

**Comment by louisevdb84** [comment]
Yes, also please run this scripts:

"migrate-grocery": "node bin/grocery/002-automigrate.js",
"grocery": "node bin/grocery/001-import.js",

Most changes was made in recipe.  I also renamed folders in the grocery file by adding the 001- and 002- in front of import and auto migrate, so that it follows recipe-search-api naming conventions. But there were no other issues that I could see on grocery

**Pull request merged**